code/controller/TranslatableCMSMainExtension.php

```diff
@@ -207,12 +207,25 @@ function MultipleLanguages() {
 
 		return (count($langs) > 1);
 	}
-	
+
 	/**
 	 * @return boolean
 	 */
 	function IsTranslatableEnabled() {
 		return SiteTree::has_extension('Translatable');
 	}
-	
+
+    /**
+     * Injects the locale into a new page on creation.
+     *
+     * @param SiteTree $record
+     * @param array $data
+     */
+    public function updateDoAdd(SiteTree $record, $data = array()) {
```

Comment:
The place in which the hook is meant for is CMSPageAddController so it does have to be a SiteTree object in practice, but yeah, this could use stdClass technically.